### DISCUSSION TITLE
Should we create an official github action?

### DISCUSSION THREAD

**Discussion by eddynaka** [spotbugs/spotbugs#1593] [General]
Hi,

I would like to check if you would like to create an official GitHub action that pushes the report to GitHub itself:

https://docs.github.com/en/code-security/secure-coding/integrating-with-code-scanning/uploading-a-sarif-file-to-github

we can help with that if you think that makes sense.

**Comment by KengoTODA** [comment]
Sorry but I couldn't catch the point.
Do you mean that the [github/codeql-action/upload-sarif](https://github.com/github/codeql-action/) isn't enough in some cases? Could you share several cases as example?

**Reply by eddynaka** [reply]
Hi @KengoTODA , the idea is to create a spotbug action where the user can send any command.
After that, we could integrate directly with github, creating a template for any user to send spotbugs file as sarif to github using the upload-sarif action.

So, imagine the following: you work in another project and you want to integrate with spotbugs.
In this case, since  spotbugs does not have an action, you would require to download the package/install the package yourself and run spotbugs.

The idea is to facilitate this scenario. So, as the consumer, I would only invoke spotbugs action in my solution and export the way I want.

something like that :)

Let me know what do u think.

**Reply by ThrawnCA** [reply]
So a person could create a GitHub Actions workflow containing something like:

    - name: SpotBugs
      uses: actions/spotbugs@v1

Could be useful in some cases, I suppose. Most tools already have a way of integrating with it, but another integration style doesn't hurt.

**Reply by eddynaka** [reply]
So, I would separate in two steps:
1. create the spotbugs action as you mentioned above
2. create a starter workflow in GitHub, which would make spotbugs appear here: https://github.com/spotbugs/spotbugs/security/code-scanning/setup

**Comment by h3xstream** [comment]
I good starting point would be have a working workflow that anybody can copy-paste.

I found two elements missing to make the SARIF integration work:
 - It was missing a text attribute.
 - It expect relative path to the **root of the repository**

The first issue was address in https://github.com/spotbugs/spotbugs/issues/1630

The second one could be done by aggregating all reports and fixing the paths after the reports are generated. Is there a better way ?

**Reply by eddynaka** [reply]
Hi @h3xstream , let us know if you need help with the fixes.
We already opened a few PRs adding a few other changes, so we can do this as well :)

**Comment by JuditKnoll** [comment]
There is also a similar github action already: https://github.com/jwgmeligmeyling/spotbugs-github-action
